db/description/collection.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package description
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/ipfs/go-datastore/query"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/core"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+func SaveCollection(
```

Comment:
suggestion: would be nice to have a short documentation for this and other methods

Reply:
A was going to point that out too. All the public functions should have documentation.

Reply:
>  All the public functions should have documentation.

These arent public, they are internal to defra

Reply:
They are very much public at the moment though. They will be accessible to users.

Reply:
I do not consider these to be public, and neither does at least one other dev.  I also think that conversation is out of scope here too.

Reply:
I strongly disagree with this one. It's either public or it's not. The functions are accessible by users and will be found in the go docs. They should be documented. If we want to keep them internal, they can be under an `internal` package.

Reply:
Discussed over discord, the `internal` package conversation will be re-opened outside of this PR.  I'll add some docs.

- [ ] Add docs for funcs